File: backend/src/main/java/com/kiwi/features/nodes/data/NodesDomain.java
package com.kiwi.features.nodes.data;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class NodesDomain {

    private Long id;
    private NodeStatus status;
    private int icon;
    private int price;
    private float cordX;
    private float cordY;
    private String name;
    private String displayName;
    private List<Long> connectedNodeIds;
    private int mapId;
    private boolean isFirstNodeOfMap;
    private String onExecutionEvent;
    private int onExecutionEntityId;
    private NodeTransitionStyle transitionStyle;
}

File: mobile/app/src/main/java/com/bellako/kiwi/features/nodes/data/NodesDomain.kt
package com.bellako.kiwi.features.nodes.data

data class NodesDomain(
    val id: Long,
    val icon: Int,
    val status: NodeStatus,
    val price: Int,
    val cordX: Float,
    val cordY: Float,
    val name: String,
    val displayName: String,
    val connectedNodeIds: List<Long>,
    val mapId: Int,
    val onExecutionEvent: String,
    val onExecutionEntityId: Int,
    val transitionStyle: NodeTransitionStyle = NodeTransitionStyle.VEIL,
)

enum class NodeStatus {
    INACCESSIBLE,
    LOCKED,
    OPEN,
    COMPLETED,
}

// Presentation hint set by content authors: declares how the entry into the
// node's follow-up sequence should be staged. Decoupled from the consequence
// of the node — the veil layer reads this directly, never inspects what the
// event ends up triggering.
enum class NodeTransitionStyle {
    VEIL,
    IMMEDIATE,
}
